server/sql.go

```diff
@@ -695,3 +709,110 @@ func validateSQLSelectStatement(sqlStatement string) bool {
 	}
 	return false
 }
+
+func (s *Server) createSQLEditorQueryActivity(ctx context.Context, c echo.Context, level api.ActivityLevel, containerID int, payload api.ActivitySQLEditorQueryPayload) error {
+	activityBytes, err := json.Marshal(payload)
+
+	if err != nil {
+		log.Warn("Failed to marshal activity after executing sql statement",
+			zap.String("database_name", payload.DatabaseName),
+			zap.String("instance_name", payload.InstanceName),
+			zap.String("statement", payload.Statement),
+			zap.Error(err))
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to construct activity payload").SetInternal(err)
+	}
+
+	activityCreate := &api.ActivityCreate{
+		CreatorID:   c.Get(getPrincipalIDContextKey()).(int),
+		Type:        api.ActivitySQLEditorQuery,
+		ContainerID: containerID,
+		Level:       level,
+		Comment: fmt.Sprintf("Executed `%q` in database %q of instance %q.",
+			payload.Statement, payload.DatabaseName, payload.InstanceName),
+		Payload: string(activityBytes),
+	}
+
+	_, err = s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{})
+
+	if err != nil {
```

Comment:
inline error checker maybe better in this situation.
```suggestion
	if _, err = s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{}); err != nil {
```